components/browser/domains/src/main/java/mozilla/components/browser/domains/DomainAutoCompleteProvider.kt

```diff
@@ -116,11 +117,14 @@ class DomainAutoCompleteProvider {
         this.useShippedDomains = useShippedDomains
 
         if (loadDomainsFromDisk) {
-            launch(UI) {
-                val domains = async(CommonPool) { Domains.load(context) }
-                val customDomains = async(CommonPool) { CustomDomains.load(context) }
-
-                onDomainsLoaded(domains.await(), customDomains.await())
+            CoroutineScope(Dispatchers.Main).launch {
+                // If a 'load' call throws, cancel the other one and re-throw.
+                coroutineScope {
+                    val domains = async { Domains.load(context) }
```

Comment:
Hm this one doesn't seem right to me. Before, we launched on the UI but had the `load`s run in a separate thread pool. Now they're in a nested scope but still run on the Main/UI thread, no?

Reply:
Yeah, good call - the new semantics are different.